app/src/main/java/com/keylesspalace/tusky/util/AccountPreferenceDataStore.kt

```diff
@@ -0,0 +1,23 @@
+package com.keylesspalace.tusky.util
+
+import androidx.preference.PreferenceDataStore
+import com.keylesspalace.tusky.db.AccountEntity
+import com.keylesspalace.tusky.settings.PrefKeys
+
+class AccountPreferenceDataStore(private val account: AccountEntity): PreferenceDataStore() {
+    override fun getBoolean(key: String?, defValue: Boolean): Boolean {
+        return when(key) {
+            PrefKeys.ALWAYS_SHOW_SENSITIVE_MEDIA -> account.alwaysShowSensitiveMedia
```

Comment:
This is not yet an exhaustive list.

But in the end it should/must handle all account settings.